cloud_functions/function.js

```diff
@@ -0,0 +1,72 @@
+const { project } = require('./config')
+const { SecretManagerServiceClient } = require('@google-cloud/secret-manager')
+const fetch = require('node-fetch')
+
+
+const flagCryptominer = async (message, context) => {
+  const data = JSON.parse(Buffer.from(message.data, 'base64').toString())
+  const token = await getSecret({ project, secretName: 'mixpanel-api' })
```

Comment:
TOL: Instead of doing this on every call, we could initialize it once in the global scope. However, that would make the secret 'stateful', and since this is probably not a high-frequency event, it might be worth keeping things simple instead of optimizing.

Reply:
I had planned to do this but forgot before creating the PR. Also wasn't sure about error handling, but I think I got that sorted out now.